MDANSE/Src/MDANSE/Framework/Configurators/GroupingLevelConfigurator.py

```diff
@@ -72,73 +73,63 @@ def configure(self, value):
 
         SingleChoiceConfigurator.configure(self, value)
 
+        self["level"] = value
+
         if value == "atom":
             return
+
         trajConfig = self._configurable[self._dependencies["trajectory"]]
         atomSelectionConfig = self._configurable[self._dependencies["atom_selection"]]
         chemical_system = trajConfig["instance"].chemical_system
         indices = []
+        flatten_indices = []
         elements = []
         names = []
         masses = []
+        group_names = []
+        group_elements = {}
+        group_n_atms = {}
         mass_lookup = chemical_system.atom_property("atomic_weight")
 
         if value == "molecule":
+            if len(trajConfig["instance"].chemical_system.unique_molecules()) == 0:
+                self.error_status = "The trajectory does not contain molecules."
+                return
+
             for mol_name in chemical_system._clusters.keys():
+                n_atms = 0
+                group_name_elements = []
                 for mol_number, cluster in enumerate(
                     chemical_system._clusters[mol_name]
                 ):
-                    indices.append(cluster)
-                    elements.append([chemical_system.atom_list[x] for x in cluster])
-                    names.append(f"{mol_name}_mol{mol_number + 1}")
-                    masses.append([mass_lookup[x] for x in cluster])
-        elif value == "group":
-            for group_name, group_indices in chemical_system._labels.items():
-                residue = set(group_indices)
-                counter = 1
-                for clustername, clusterlist in chemical_system._clusters.items():
-                    for cluster in clusterlist:
-                        molecule = set(cluster)
-                        if molecule.issubset(residue) or residue.issubset(molecule):
-                            indices.append(list(molecule.intersection(residue)))
-                            elements.append(
-                                [chemical_system.atom_list[x] for x in cluster]
-                            )
-                            names.append(f"{group_name}_num{counter}_in_{clustername}")
-                            masses.append([mass_lookup[x] for x in cluster])
-                            counter += 1
+                    for x in cluster:
+                        if x not in atomSelectionConfig["flatten_indices"]:
+                            continue
+                        indices.append([x])
+                        flatten_indices.append(x)
```

Comment:
I think this can be simplified, `more_itertools` provides the [`collapse`](https://more-itertools.readthedocs.io/en/stable/api.html#more_itertools.collapse) iterator which should iterate over the flattened indices.